FWCore/Framework/src/ProductResolvers.h

```diff
@@ -40,6 +40,7 @@ namespace edm {
 
     // Give AliasProductResolver and SwitchBaseProductResolver access by moving this method to public
     void resetProductData_(bool deleteEarly) override = 0;
+    virtual ProductData const& getProductData() const = 0;
```

Comment:
I'm not very happy about exposing the `ProductData` this way. It is used here
https://github.com/cms-sw/cmssw/blob/9890e15de9fa531e391ffe4d0f5e6a3cfc6669c8/FWCore/Framework/src/ProductResolvers.cc#L659-L662
to set the Wrapper in the prefetching phase.